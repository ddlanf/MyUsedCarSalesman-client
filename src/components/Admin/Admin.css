
.admin{
   z-index: 15;
   position: relative;
   margin-top: -2%;
   background-color: white;
}


.admin-header{
   padding-top: 5%;
   font-size: 5vw;
}

.admin-info{
    margin-left: auto;
    margin-right: auto;
    background-color: lightgray;
}
 
 .admin-info > *{
    font-size: 2vw;
 }

 .admin-paragraph{
   font-size: 1vw;
 }
 
 .admin-table{
    overflow-y:scroll;
    overflow-x:scroll;
    height: 30vw;
    display:block;
    text-align: left;
    border-collapse: collapse;
    margin-left: 0.2%;
 }
 
th{
   border: 1px solid black;
 }

.admin-delete, .admin-block{
   width: 10vw;
   font-size: 2vw;
 }
 
 .admin-return{
   margin-top: 5%;
   margin-left: auto;
   margin-right: auto;
   margin-bottom: 10%;
   display: block;
   border: black solid 2px;
   width: 20%;
   text-align: center;
   padding-top: 1%;
   background-color: yellow;
   height: 5vw;
   font-size: 2vw;
 }
 
 .admin-return:hover{
   background-color: brown;
 }
 