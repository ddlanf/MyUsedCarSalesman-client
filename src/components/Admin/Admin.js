import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import './Admin.css'
import UserInfo from './UserInfo/UserInfo'
import UserPosts from './UserPosts/UserPosts'
import UserComplaintsAndReports from './UserComplaintsAndReports/UserComplaintsAndReports'
import UserService from '../../services/users-api-service'
import PostService from '../../services/posts-api-service'
import ReportApiService from '../../services/report-api-service'
export default class Admin extends Component {

    constructor(props){
        super(props)
        this.state ={
            users:[],
            posts: [],
            reports:[]
        }
    }
    
    componentDidMount(){
        UserService.getUsers()
            .then(users => this.setState({users}))
        PostService.getPosts()
            .then(posts => this.setState({posts}))
        ReportApiService.getReports()
            .then(reports => this.setState({reports}))
    }

    updateUsers = () => {
        UserService.getUsers()
            .then(users => this.setState({users}))
    }

    updatePosts = () => {
        PostService.getPosts()
            .then(posts => this.setState({posts}))
    }

    adminExit = () =>{
        this.props.userLogOut()
    }

    render() {
        return (
            <div className="admin-background">
                <div className="admin">
                    <h1 className="admin-header">Admin Info</h1>
                    <section className="admin-info">   
                        <UserInfo users={this.state.users} updateUsers={this.updateUsers}/>
                    </section>

                    <section className="admin-info">
                        <UserComplaintsAndReports reports={this.state.reports}/>
                    </section>

                    <section className="admin-info">
                        <UserPosts
                            users={this.state.users} 
                            posts={this.state.posts}
                            updatePosts={this.updatePosts}
                            />
                    </section>
                    <Link
                        to={'/view-posts'}
                        >
                        <button
                            className='admin-return'
                            onClick={this.adminExit}
                            >
                            Return
                        </button>
                    </Link>
                </div>
            </div>
        )
    }
}
