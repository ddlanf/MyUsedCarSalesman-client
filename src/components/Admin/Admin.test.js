import React from 'react';
import ReactDOM from 'react-dom';
import Admin from './Admin';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'

describe('Admin component', () => {
    
    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                        <Admin/>
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(
                <BrowserRouter>
                    <Admin/>
                </BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});