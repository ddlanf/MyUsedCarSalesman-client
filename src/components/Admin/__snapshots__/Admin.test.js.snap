// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin component renders the UI as expected 1`] = `
<div
  className="admin"
>
  <h1
    className="admin-header"
  >
    Admin Info
  </h1>
  <section
    className="admin-info"
  >
    <div
      className="users-info users"
    >
      <h2>
        User Info
      </h2>
      <table
        className="admin-table users"
      >
        <thead>
          <tr
            className="about-users"
          >
            <th>
              ID
            </th>
            <th>
              Username
            </th>
            <th>
              Password
            </th>
            <th>
              First name
            </th>
            <th>
              Last name
            </th>
            <th>
              Email
            </th>
            <th>
              Date Created
            </th>
            <th />
            <th />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </section>
  <section
    className="admin-info"
  >
    <div
      className="users-info reports"
    >
      <h2>
        User Complaints and reports
      </h2>
      <table
        className="admin-table reports"
      >
        <thead>
          <tr
            className="about-reports"
          >
            <th>
              ID
            </th>
            <th>
              Message
            </th>
            <th>
              Type
            </th>
            <th>
              User ID
            </th>
            <th>
              Date Sent
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </section>
  <section
    className="admin-info"
  >
    <div
      className="users-info posts"
    >
      <h2>
        User Posts
      </h2>
      <table
        className="admin-table posts"
      >
        <thead>
          <tr
            className="about-posts"
          >
            <th>
              ID
            </th>
            <th>
              Make
            </th>
            <th>
              Model
            </th>
            <th>
              Year
            </th>
            <th>
              Mileage
            </th>
            <th>
              Description
            </th>
            <th>
              Price
            </th>
            <th>
              Commission Amount
            </th>
            <th>
              Location
            </th>
            <th>
              Other terms and conditions
            </th>
            <th>
              User ID
            </th>
            <th />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </section>
  <a
    className="admin-return"
    href="/view-posts"
    onClick={[Function]}
  >
    Return
  </a>
</div>
`;
